### PR TITLE
fix(LoginPages): add retry in forgot password and signup

### PR DESCRIPTION
**Pull request by fcoronelmakingsense**
- Add retry to signup page and forgot password. 
- Refactor to general functionality to add more urls in the future if we need more.

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -203,3 +203,12 @@ const urlsWebApp: any = [
...
 }
+
+export function findUrlInRetryList(currentUrl: any): any {
+  const urls = [
+    'WebAppPublic/Login',
+    'WebAppPublic/CreateUser',
+    '/WebAppPublic/SendResetPasswordEmail',
+  ];
+  return urls.find((item: string) => item === currentUrl);
+}
```
Comment (andresmoschini) [inline src/utils.ts:214]: I think that these URLs are very tied to legacy Doppler, what do you think about keep it inside doppler-legacy-client.ts?

File: src/utils.ts [hunk 2]
```diff
@@ -203,3 +203,12 @@ const urlsWebApp: any = [
 export function getCurrentPageForUrl(currentUrl: string): any {
   return urlsWebApp.find((item: any) => item.url === currentUrl);
 }
+
+export function findUrlInRetryList(currentUrl: any): any {
+  const urls = [
+    'WebAppPublic/Login',
+    'WebAppPublic/CreateUser',
+    '/WebAppPublic/SendResetPasswordEmail',
```
Comment (andresmoschini) [inline src/utils.ts:211]: Why this one has a different format?

File: src/utils.ts [hunk 3]
```diff
@@ -203,3 +203,12 @@ const urlsWebApp: any = [
 export function getCurrentPageForUrl(currentUrl: string): any {
   return urlsWebApp.find((item: any) => item.url === currentUrl);
 }
+
+export function findUrlInRetryList(currentUrl: any): any {
```
Comment (lucianaDilizio) [inline src/utils.ts:207]: `(currentUrl: any)` why not `string`?

Comment (fcoronelmakingsense) [reply src/utils.ts:207]: can be null

File: src/utils.ts (hunk 2 above)
Comment (fcoronelmakingsense) [reply src/utils.ts:211]: because in doppler legacy has the `/` maybe need delete both sides

Comment (andresmoschini) [reply src/utils.ts:211]: And it is different in the other cases?

File: src/utils.ts (hunk 3 above)
Comment (andresmoschini) [reply src/utils.ts:207]: In that case:

```suggestion
export function  isUrlInRetryList(currentUrl: string | null): bool {
```

File: src/utils.ts (hunk 2 above)
Comment (fcoronelmakingsense) [reply src/utils.ts:211]: yes other cases has the `/` too. maybe after can add refactor to all urls

File: src/utils.ts (hunk 3 above)
Comment (fcoronelmakingsense) [reply src/utils.ts:207]: but if find the url return the element dont a boolean

Comment (andresmoschini) [reply src/utils.ts:207]: > but if find the url return the element dont a boolean

I know that you can fix that :P

As I understand, you do not want to find a URL, you want to verify if it is in the list, for that reason I purpose to change the name of the function and the result type.

File: src/services/doppler-legacy-client.ts [hunk 4]
```diff
@@ -341,7 +350,6 @@ export class HttpDopplerLegacyClient implements DopplerLegacyClient {
       retryDelay: (retryCount: any, error: any) => {
         addLogEntry({
           origin: window.location.origin,
-          section: 'Login/legacyClient',
```
Comment (andresmoschini) [inline src/services/doppler-legacy-client.ts:344]: Is it not necessary anymore?

File: src/services/doppler-legacy-client.ts [hunk 5]
```diff
@@ -312,8 +312,17 @@ export class HttpDopplerLegacyClient implements DopplerLegacyClient {
...
   private window: any;
 
+  private findUrlInRetryList(currentUrl: any): any {
+    const urls = [
+      'WebAppPublic/Login',
+      'WebAppPublic/CreateUser',
+      'WebAppPublic/SendResetPasswordEmail',
+    ];
+    return urls.find((item: string) => item === currentUrl);
+  }
```
Comment (andresmoschini) [inline src/services/doppler-legacy-client.ts:322]:
```suggestion
  private findUrlInRetryList(currentUrl: string | null): bool {
    const urls = [
      'WebAppPublic/Login',
      'WebAppPublic/CreateUser',
      'WebAppPublic/SendResetPasswordEmail',
    ];
    return urls.includes(currentUrl);
  }
```

File: src/services/doppler-legacy-client.ts (hunk 4 above)
Comment (cbernat) [reply src/services/doppler-legacy-client.ts:344]: We cannot have all logs with the same section, either remove it or replace it. Also this informatiton is a bit redundant, see in the log:

image

Here we have section and then in request information we have also the url.

File: src/utils.ts (hunk 1 above)
Comment (fcoronelmakingsense) [reply src/utils.ts]: fixed

File: src/utils.ts (hunk 3 above)
Comment (fcoronelmakingsense) [reply src/utils.ts]: fixed

File: src/services/doppler-legacy-client.ts (hunk 5 above)
Comment (fcoronelmakingsense) [reply src/services/doppler-legacy-client.ts]: fixed

**Pull request merged**